src/main/java/org/apache/mesos/state/StateStore.java

```diff
@@ -207,4 +207,9 @@
      */
     void clearProperty(final String key) throws StateStoreException;
 
+    /**
+     * Provides the list of Resources which are expected to be reserved to a particular role.
+     * @return
+     */
+    List<Protos.Resource> getExpectedResources();
```

Comment:
As mentioned before it feels like this would be better placed as a helper method upstream of StateStore.